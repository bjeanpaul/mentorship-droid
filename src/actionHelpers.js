import { partialRight as partial, isFunction } from 'lodash';
import { switchError } from 'src/helpers';


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type,
});


export const dataAction = type => data => ({
  type,
  payload: data,
});


export const castThunk = fn => (...args) => {
  const obj = fn(...args);
  return !isFunction(obj)
    ? dispatch => dispatch(obj)
    : obj;
};
