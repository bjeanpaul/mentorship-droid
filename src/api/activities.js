import { arrayOf } from 'normalizr';
import request from 'src/api/request';
import { Activity } from 'src/api/schemas';
import { parseResults } from 'src/api/parse';


export const listActivities = auth => request({
  url: '/activity/',
  method: 'GET',
  schema: arrayOf(Activity),
  parse: parseResults,
  auth,
});


export const getActivity = (id, auth) => request({
  url: `/activity/${id}/`,
  method: 'GET',
  schema: Activity,
  auth,
});


export const updateActivity = (id, data, auth) => request({
  url: `/activity/${id}/`,
  method: 'PUT',
  data,
  auth,
});
