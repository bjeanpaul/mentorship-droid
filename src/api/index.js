export * from 'src/api/schemas';
export * from 'src/api/profiles';
export * from 'src/api/schedule';
export * from 'src/api/activities';
export * from 'src/api/categories';
export * from 'src/api/events';
export * from 'src/api/errors';
export * from 'src/api/sync';
export * from 'src/api/calls';
export * from 'src/api/notificationSettings';
export * from 'src/api/callNotes';
export * from 'src/api/messages';
export * from 'src/api/imageUrl';
export { default as imageUrl } from 'src/api/imageUrl';
