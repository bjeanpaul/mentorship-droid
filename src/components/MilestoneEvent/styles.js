import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.EVENT_BG,
    marginTop: 12,
    margin: 24,
    padding: 20,
    paddingLeft: 16,
    paddingRight: 16,
    borderRadius: 4,
  },
  imageContainer: {
    alignSelf: 'center',
    backgroundColor: colors.EVENT_MILESTONE_IMAGE_BG,
    width: 112,
    height: 112,
    borderRadius: 66,
    overflow: 'hidden',
    marginBottom: 20,
  },
  milestonReachedText: {
    fontSize: 16,
    color: colors.EVENT_MILESTONE_REACHED_TEXT,
  },
  categoryNameCompletedText: {
    fontSize: 24,
    color: colors.EVENT_MILESTONE_CATEGORY_NAME_TEXT,
  },
});
