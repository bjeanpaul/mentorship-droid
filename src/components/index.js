import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
