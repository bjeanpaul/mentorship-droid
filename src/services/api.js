const HOSTNAME = 'http://localhost:8000/mentor'

export function callAPI(endpoint, hash) {

  return fetch(`${HOSTNAME}${endpoint}`, {
    headers: {
      'Authorization': `Basic ${hash}`
    }
  })
  .then(response => response.json().then(json => ({ json, response })))
  .then(({json, response}) => {
    if (response.ok === false) {
      return Promise.reject(json)
    }
    return json
  })

}
