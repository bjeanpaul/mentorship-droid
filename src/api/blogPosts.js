import { arrayOf } from 'normalizr';

import request from 'src/api/request';
import { parseBlogPostListResults } from 'src/api/parse';
import { BlogPost } from 'src/api/schemas';


export const listBlogPosts = (auth, params = {}) => request({
  url: '/blog/',
  method: 'GET',
  schema: arrayOf(BlogPost),
  parse: parseBlogPostListResults,
  params,
  auth,
});
