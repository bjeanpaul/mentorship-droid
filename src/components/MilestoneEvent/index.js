import React, { PropTypes } from 'react';
import { View, Image } from 'react-native';

import { ImageUrl } from 'src/api';
import { Text } from 'src/components';
import styles from './styles';


const MilestoneEvent = ({
  image,
  title,
  color,
}) => (
  <View style={[styles.container, { backgroundColor: color }]}>
    <View style={styles.imageContainer}>
      <Image source={image.toSource()} />
    </View>
    <Text style={styles.milestonReachedText}>Milestone Reached</Text>
    <Text style={styles.categoryNameCompletedText}>{title} Completed</Text>
  </View>
);


MilestoneEvent.propTypes = {
  image: PropTypes.instanceOf(ImageUrl).isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
};


export default MilestoneEvent;
