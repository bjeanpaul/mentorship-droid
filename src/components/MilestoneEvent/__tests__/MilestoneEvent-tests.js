import React from 'react';
import MilestoneEvent from 'src/components/MilestoneEvent';
import { imageUrl } from 'src/api';


describe('MilestoneEvent', () => {
  const createComponent = (props = {}) => (
    <MilestoneEvent
      image={imageUrl('/foo.jpg')}
      title="pew pew pew"
      color="blue"
      {...props}
    />
  );

  it('should render the props', () => {
    expect(render(createComponent()).toJSON()).toMatchSnapshot();
  });

  it('should not render the category image if it does not exist', () => {
    expect(render(createComponent({
      image: imageUrl(null),
    })).toJSON()).toMatchSnapshot();
  });
});
