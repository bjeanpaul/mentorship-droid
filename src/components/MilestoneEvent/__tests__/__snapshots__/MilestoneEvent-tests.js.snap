exports[`MilestoneEvent should not render the category image if it does not exist 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 4,
        "margin": 24,
        "marginTop": 0,
        "padding": 20,
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.35)",
          "fontSize": 16,
        },
      ]
    }>
    Milestone Reached
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center",
        },
        Object {
          "color": "#fff",
          "fontSize": 24,
        },
      ]
    }>
    pew pew pew
     Completed
  </Text>
</View>
`;

exports[`MilestoneEvent should render the props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 4,
        "margin": 24,
        "marginTop": 0,
        "padding": 20,
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 66,
        "marginBottom": 20,
      }
    }>
    <Image
      source={
        Object {
          "uri": "/api/foo.jpg?filter_spec=min-224x224",
        }
      }
      style={
        Object {
          "borderRadius": 66,
          "height": 112,
          "width": 112,
        }
      } />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.35)",
          "fontSize": 16,
        },
      ]
    }>
    Milestone Reached
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center",
        },
        Object {
          "color": "#fff",
          "fontSize": 24,
        },
      ]
    }>
    pew pew pew
     Completed
  </Text>
</View>
`;
